### PR TITLE
chore: smart cdk-nag unit tests on warnings and failures

### PR DESCRIPTION
**Pull request by kcadette**
Issue #, if available:

Description of changes:

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/cdk-infra.test.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            patientPickerEndpoint,
+            enableBackup,
+            fhirVersion,
+            description:
+                'Test - Primary Template - This template creates all the necessary resources to deploy FHIR Works on AWS; a framework to deploy a FHIR server on AWS. It will be used to run cdk-nag.',
+        });
+
+        cdk.Aspects.of(stack).add(new cdknag.AwsSolutionsChecks({ verbose: true }));
+
+        cdknag.NagSuppressions.addStackSuppressions(stack, [
```
Comment (Bingjiling) [inline src/test/cdk-infra.test.ts:52]: Can we drill down to specific resources that causes the warning/error, and suppress accordingly?

Comment (kcadette) [reply src/test/cdk-infra.test.ts:52]: *disclaimer: I did not include these suppressions, they are already included in bin/cdk-infra.ts. 
But I did look into them and there over 100 resources that trigger these rules and the reason explains that it's by design

File: .github/workflows/cfn-analysis.yml [hunk 2]
```diff
@@ -26,21 +26,11 @@ jobs:
           node-version: 14
       - name: Install npm dependencies
         run: yarn install
-      - name: Package main sls into cfn for analysis
+      - name: synthesize cdk template for cfn for analysis
         run: |
-          yarn run serverless-package --stage dev --region us-west-2 --useHapiValidator true --conceal
-      - name: cfn_nag on main serverless
+          yarn cdk synth -c issuerEndpoint='test' -c oAuth2ApiEndpoint='test' -c patientPickerEndpoint='test'
```
Comment (ssvegaraju) [inline .github/workflows/cfn-analysis.yml]: do we need `--require-approvals=never` here? Not sure if synth also produces a confirmation message if there are security changes or not.

File: .github/workflows/cfn-analysis.yml [hunk 3]
```diff
@@ -26,21 +26,11 @@ jobs:
           node-version: 14
       - name: Install npm dependencies
         run: yarn install
-      - name: Package main sls into cfn for analysis
+      - name: synthesize cdk template for cfn for analysis
         run: |
-          yarn run serverless-package --stage dev --region us-west-2 --useHapiValidator true --conceal
-      - name: cfn_nag on main serverless
+          yarn cdk synth -c issuerEndpoint='test' -c oAuth2ApiEndpoint='test' -c patientPickerEndpoint='test' -c require-approval=never
```
Comment (ssvegaraju) [inline .github/workflows/cfn-analysis.yml]: Quick thing: this should be `--require-approval=never`, not `-c require-approval=never`. Other than that PR looks good to me!

**Review by ssvegaraju — APPROVED** [review]

**Pull request merged**